packages/react-router/src/link.tsx

```diff
@@ -545,6 +545,7 @@ export function useLinkProps<
   options: UseLinkPropsOptions<TRouter, TFrom, TTo, TMaskFrom, TMaskTo>,
 ): React.AnchorHTMLAttributes<HTMLAnchorElement> {
   const router = useRouter()
+  const isClicking = React.useRef(false)
```

Comment:
why do we need to track if a link was clicked?
what happens if we have two link instances pointing to the same target, this then won't work, right?

i think the solution must be independent from click tracking etc, and also must work when programatically preloading / navigating without clicking on a link

Reply:
this most likely needs to be fixed inside of `router.ts`

Reply:
This is a different issue than the references (so we could have it in a separate PR), but essentially `onFocus` is also invoked when a button is clicked, which causes the route to get preloaded twice (`onMouseEnter` + `onFocus`).

When focusing a link via keyboard navigation, we obviously skip `onMouseEnter` so it gets correctly preloaded once.
Given that there's no way to distinguish between mouse and keyboard origin for focus events, we can "trap" it by skipping the preload call when a focus event is detected between `onMouseDown` and `onMouseUp`: this makes preload only happen once and ensures that the behaviour is the same for both mouse and keyboard navigation.

Reply:
ok then please put this in a separate PR then

Reply:
https://github.com/TanStack/router/pull/1885